src/mgmt/srv_query.c

```diff
@@ -761,7 +761,23 @@ ds_mgmt_dev_manage_led(Ctl__LedManageReq *req, Ctl__DevManageResp *resp)
 	ctl__smd_device__init(resp->device);
 	resp->device->uuid = NULL;
 
-	led_info.tr_addr  = req->ids;
+	D_ALLOC_PTR(resp->device->ctrlr);
+	if (resp->device->ctrlr == NULL) {
+		return -DER_NOMEM;
+	}
+	ctl__nvme_controller__init(resp->device->ctrlr);
+
+	D_ALLOC(resp->device->ctrlr->pci_addr, ADDR_STR_MAX_LEN + 1);
+	if (resp->device->ctrlr->pci_addr == NULL)
+		return -DER_NOMEM;
+	if ((req->ids == NULL) || (strlen(req->ids) == 0)) {
```

Comment:
We should be avoiding strlen. Could we use strnlen with ADDR_STR_MAX_LEN?

Reply:
I will address when I have to re-push (after the parent lands).